src/FindNetworkEndpoint/index.ts

```diff
@@ -0,0 +1,39 @@
+import { INetwork } from './interface';
+
+export class FindNetworkEndpoint {
+  network: INetwork;
+  startNodeId: number;
+
+  constructor(startNodeId: number, fromIds: number[], toIds: number[]) {
+    this.network = {
+      fromIds,
+      toIds
+    };
```

Comment:
I think it is worth to parse it here for easier path finding, so let's say you have these 2 arrays, as you know the graph  has only one way out from each node then you can use simple JS object, so if you have arrays like this
```
        const fromIds = [1, 7, 3, 4, 2, 6, 9];
        const toIds = [3, 3, 4, 6, 6, 9, 5];
``` 
then you can parse it to this object
```
const graph = {
 1:  3,
 7: 3,
 3: 4,
 4: 6,
 2: 6,
 6: 9,
 9: 5
}
```

then you wouldn't need to do any array finding

Reply:
It is a better idea.